docs/architecture/adr-template.md

```diff
@@ -0,0 +1,15 @@
+# ADR 000: Template for an ADR
```

Comment:
Do you think the template I had [here](https://github.com/cosmos/cosmos-sdk/pull/1882/commits/6d75b69a5c937814b700a830045bdb710a5c8f67#diff-5be5997d8f94baad927ca518d17f9417) would be more helpful/descriptive?

Reply:
Oh cool didn't see that! Yeah that template looks way better, I'll copy it here

Reply:
✊